openedx/core/djangoapps/discussions/url_helpers.py

```diff
@@ -54,3 +54,20 @@ def get_discussions_mfe_topic_url(course_key: CourseKey, topic_id: str, view: Op
         (str) URL link for MFE. Empty if the base url isn't configured
     """
     return _get_url_with_view_query_params(f"{course_key}/topics/{topic_id}", view)
+
+def get_discussions_mfe_topic_thread_url(course_key: CourseKey, topic_id: Optional[str] = None, thread_id: Optional[str] = None, view: Optional[str] = None) -> str:
+    """
+    Returns the url for discussions for the specified course and topic and thread in the discussions MFE.
+    Args:
+        course_key (CourseKey): course key of course for which to get url
+        topic_id (str): topic id for topic to get url for
+        thread_id (str): thread id for topic to get url for
+        view (str): which view to generate url for
+    Returns:
+        (str) URL link for MFE. Empty if the base url isn't configured
+    """
+    if thread_id and topic_id:
+        return _get_url_with_view_query_params(f"{course_key}/topics/{topic_id}/posts/{thread_id}", view)
+    elif topic_id:
+        return _get_url_with_view_query_params(f"{course_key}/topics/{topic_id}", view)
+    return get_discussions_mfe_url(course_key,view)
```

Comment:
I think instead of creating a new function, you can rename the original function and overload it to support this since it covers both use cases.